nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -86,6 +88,11 @@ public static void checkDocumentEventContext(Event event, DocumentModel expected
 
         assertTrue(properties.containsKey(COMMENT));
         assertEquals(expectedCommentDocModel.getPropertyValue(COMMENT_TEXT), properties.get(COMMENT));
+
+        // The event source document must be the top level document as is the one linked in the notification.
+        DocumentModel topLevelDocument = (DocumentModel) properties.get(TOP_LEVEL_DOCUMENT);
+        DocumentModel sourceDoc = ((DocumentEventContext) event.getContext()).getSourceDocument();
+        assertEquals(topLevelDocument.getId(), sourceDoc.getId());
```

Comment:
The assertion we want is the one below:
```java
assertTrue(properties.containsKey(TOP_LEVEL_DOCUMENT));
DocumentModel topLevelDocument = (DocumentModel) properties.get(TOP_LEVEL_DOCUMENT);
assertEquals(expectedCommentedDocModel.getRef(), topLevelDocument.getRef());
```
And eventually add at the beginning of the method (I didn't had a look if such assertion is done elsewhere):
```java
DocumentModel sourceDoc = ((DocumentEventContext) event.getContext()).getSourceDocument();
assertEquals(expectedCommentDocModel.getRef(), sourceDoc.getRef());
```

Reply:
We can't systematically check in the same method
`assertEquals(expectedCommentedDocModel.getRef(), topLevelDocument.getRef());`
&&
`assertEquals(expectedCommentDocModel.getRef(), sourceDoc.getRef());`

as sometimes the `sourceDoc` is not the `topLevelDocument` (in the case of reply).

I only add the first check as this one needs to be always true.

Reply:
`expectedCommentedDocModel` is the parent of the comment.
`expectedCommentedDocModel` is not the `topLevelDocument` when you make a reply.
It didn't show until now because the tests are written with only 1 level of comments.

In the case of a reply we distinguish the `topLevelDocument` from `expectedCommentedDocument` but there is no room for this in the methods as they are.